src/ACEtk/Xsdir/src/parse.hpp

```diff
@@ -22,46 +22,70 @@ static void trimToLowerCase( std::string& string ) {
   toLowerCase( string );
 }
 
+static std::string getline( std::istream& in, std::istream::pos_type& position ) {
+
+  // a line is a comment line if the first character on it is '#'
+  auto isCommentLine = [] ( const auto& line ) {
+
+    if ( line.size() > 0 ) {
+
+      return line[0] == '#' ? true : false;
```

Comment:
This line is identical to `return line[0] == '#'`. I don't imagine this is more particularly readable. Though, I admit I'd rewrite this as:
```cpp
auto isCommentLine = [] (const auto& line) {
    return (line.size() > 0 && line[0] == '#');
}
```